### ISSUE TITLE
Dom nodes memory leak with animations

### ISSUE THREAD

**Issue by nicolashenry** [angular/angular#16239]
**I'm submitting a ...**
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
I note while profiling my application that  I have detached dom nodes after an animated elements have been removed from dom tree.

**Expected behavior**
Detached node should be flushed from memory.

**Minimal reproduction of the problem with instructions**
I made an example here:
http://plnkr.co/edit/Ol6JW8T3Stp8X4hcw7v9?p=preview

Each time I click on "Generate items", it create 1000 animated dom nodes which are removed just after.
With Chrome developer tools I see detached dom tree in heap profiling:
![image](https://cloud.githubusercontent.com/assets/139160/25281948/4f02bdd2-26af-11e7-8896-fd7856b973a6.png)
I think that the _elementTriggerStates map keys in [dom_animation_engine.ts](https://github.com/angular/angular/blob/4.0.x/packages/animations/browser/src/render/dom_animation_engine.ts) are never removed from the map which causes this problem.

**What is the motivation / use case for changing the behavior?**
I have some memory crash causes by this on my cordova android mobile application (mostly due to img elements keeped in memory).

**Please tell us about your environment:**

* **Angular version:** 4.0.3

* **Browser:** all
 
* **Language:** all

**Comment by laco0416** [comment]
I have same situation w/ `@angular/animations` 4.1.1

**Comment by nicolashenry** [comment]
This issue seems to be fixed in 4.2.x
